geo/src/test/java/org/odk/collect/geo/GeoPointActivityTest.java

```diff
@@ -0,0 +1,165 @@
+package org.odk.collect.geo;
+
+import static android.app.Activity.RESULT_OK;
+import static android.provider.Settings.ACTION_LOCATION_SOURCE_SETTINGS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.robolectric.Shadows.shadowOf;
+
+import android.content.Intent;
+import android.location.Location;
+
+import androidx.lifecycle.Lifecycle;
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.odk.collect.location.LocationClient;
+import org.odk.collect.location.LocationClientProvider;
+
+@RunWith(AndroidJUnit4.class)
+public class GeoPointActivityTest {
+
+    LocationClient locationClient = mock(LocationClient.class);
+
+    @Before
+    public void setUp() throws Exception {
+        LocationClientProvider.setTestClient(locationClient);
+    }
+
+    @Test
+    public void testLocationClientLifecycle() {
```

Comment:
I'd like to rewrite this test, but it felt like moving these files around was enough of a time sink. Another day...